Docs updates for Cloud - Jan 22 (#29558)
sourcegraph/sourcegraph: doc/cloud/organizations/code_visibility_teams_sourcegraph_cloud.md

## Public repositories

> If a repository is public on GitHub.com or GitLab.com, other users on Sourcegraph Cloud can view and search across that repository. The repository will appear in the global search context. If a public repository is moved to private on the code host, Sourcegraph will respect the updated permissions. 

Reviewer: ```suggestion
If a repository is public on GitHub.com or GitLab.com, other users on Sourcegraph Cloud can view and search across that repository. The repository will appear in the global search context. If a public repository is made private on the code host, that repository will no longer appear in the global search context. 
```

## Private repositories

> TL;DR: Users on Sourcegraph can only see a private repository if they have permissions to see that repository on your code host and if they are a member of the organization on Sourcegraph. 

Reviewer: ```suggestion
TL;DR: Users on Sourcegraph can only see a private repository if they have permissions to see that repository on your code host and if they are a member of your organization on Sourcegraph. 
```
